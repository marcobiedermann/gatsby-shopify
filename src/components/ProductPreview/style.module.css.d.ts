declare const styles: {
  readonly 'productPreview': string;
};
export = styles;
