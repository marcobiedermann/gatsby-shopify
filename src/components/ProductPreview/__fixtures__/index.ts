import { ProductPreviewProps } from '..';

const productPreview: ProductPreviewProps = {
  handle: 'product',
  featuredImage: {
    originalSrc: '/path/to/image.jpg',
  },
  priceRangeV2: {
    minVariantPrice: {
      amount: '100',
      currencyCode: 'EUR',
    },
  },
  shopifyId: '12345678-9012-3456-7890-123456789012',
  title: 'Product Title',
};

export default productPreview;
