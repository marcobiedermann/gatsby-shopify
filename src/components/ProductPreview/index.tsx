import { Link } from 'gatsby';
import React, { FC } from 'react';
import Image from '../Image';
import * as styles from './style.module.css';

interface Image {
  originalSrc: string;
}

interface PriceRange {
  minVariantPrice: VariantPrice;
}

interface VariantPrice {
  amount: string;
  currencyCode: string;
}

export interface ProductPreviewProps {
  handle: string;
  images: Image[];
  priceRange: PriceRange;
  shopifyId: string;
  title: string;
}

const ProductPreview: FC<ProductPreviewProps> = (props) => {
  const { handle, images, title, priceRange } = props;

  return (
    <div className={styles.productPreview}>
      <figure>
        <Link to={`/products/${handle}`}>
          <Image src={images[0].originalSrc} alt={title} />
        </Link>
      </figure>
      <h3>
        <Link to={`/products/${handle}`}>{title}</Link>
      </h3>
      <p>
        {priceRange.minVariantPrice.amount} {priceRange.minVariantPrice.currencyCode}
      </p>
    </div>
  );
};

export default ProductPreview;
