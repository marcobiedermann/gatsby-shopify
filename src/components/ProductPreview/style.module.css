.product-preview {
  padding: 2em;
}
