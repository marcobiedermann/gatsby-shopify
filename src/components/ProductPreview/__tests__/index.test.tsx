import { render } from '@testing-library/react';
import React from 'react';
import ProductPreview from '..';
import productPreview from '../__fixtures__';

describe('product preview component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<ProductPreview {...productPreview} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
