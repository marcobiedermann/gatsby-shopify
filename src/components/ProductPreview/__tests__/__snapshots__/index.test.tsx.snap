// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product preview component should render correctly 1`] = `
<div>
  <figure>
    <a
      href="/products/product"
    >
      <img
        alt="Product Title"
        src="/path/to/image.jpg"
      />
    </a>
  </figure>
  <h3>
    <a
      href="/products/product"
    >
      Product Title
    </a>
  </h3>
  <p>
    100
     
    EUR
  </p>
</div>
`;
